command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -39,17 +39,42 @@ export class Command<A extends ArgType[] | { [argName: string]: any } = [] | {}>
     }
   }
 
+  private buildSeqTime(time: Temporal.Instant | Temporal.Duration): string {
```

Comment:
I think a pro was combining the time-building logic into one method that can be used by different time objects. We can expand this method to accept another time format in the future. It is also compact. A con that isn't really that bad is to have two or more time-building methods so slightly more code with method declaration.  

If the time conversion logic was really complicated then I would have split it into two methods but I felt it wasn't too bad to read.

Reply:
Now that matt mentions it, I agree with him. I think this generic time building function is pushing context about the time format down into a function that shouldn't know about that context.

The place its called (toSeqJson) knows what formats it needs and should request those formats rather than delegating that knowledge elsewhere.

I'd recommend having two functions - durationToHms and instantToDoy

Reply:
Sounds good, pretty easy change on my end, I retested and pushed the changes to the PR